workspaces/redhat-argocd/plugins/argocd/src/components/DeploymentLifeCycle/sidebar/resources/ResourcesTable.tsx

```diff
@@ -69,24 +75,41 @@ export const ResourcesTable: FC<ResourcesTableProps> = ({
     [resources],
   );
 
+  const getAllKinds = React.useMemo(() => {
+    return resources.reduce((kinds: string[], resource) => {
+      if (resource.kind && !kinds.includes(resource.kind)) {
+        kinds.push(resource.kind);
+      }
+      return kinds;
+    }, []);
+  }, [resources]);
+
   const filteredResources = React.useMemo(() => {
     let items = allResources;
 
-    // Filter by health status
-    if (filterValue && filterValue !== 'All') {
-      items = items.filter(resource => resource.health.status === filterValue);
+    // Filters
+    if (filters.HealthStatus.length) {
+      items = items.filter(resource =>
+        filters.HealthStatus.includes(resource.health.status),
```

Comment:
Using Health filters is crashing the App
```
Cannot read properties of undefined (reading 'status')
 ```
```suggestion
        filters.HealthStatus.includes(resource.health?.status),
```

Reply:
Also can you update the type `Resource` type to mark this health as optional field ?